File: PYTHON/CartesianPoint/point.py
import math


#Global function
def FarthestPoint(points_list):
  index_farthest_point = 0
  for index in range(0, len(points_list)):
    if(points_list[index].GetDistanceFromOrigin() > points_list[index_farthest_point].GetDistanceFromOrigin()):
      index_farthest_point = index
  
  return index_farthest_point


class Point:
  
  def __init__(self, x, y):
    self.__x = x
    self.__y = y
    self.__distance = math.sqrt(pow(x, 2) + pow(y, 2))
    
  def GetX(self):
    return self.__x

  def GetY(self):
    return self.__y

  def GetDistanceFromOrigin(self):
    return self.__distance

  def SetX(self, x):
    self.__x = x
    self.SetDistanceFromOrigin()
    
  def SetY(self, y):
    self.__y = y
    self.SetDistanceFromOrigin()

  def SetDistanceFromOrigin(self):
    self.__distance = math.sqrt(pow(self.__x, 2) + pow(self.__y, 2))

  def DisplayCoordinates(self):
    print "(" + str(self.GetX()) + ", " + str(self.GetY()) + ")"
  
  def GetDistanceFromPoint(self, other_point):
    return math.sqrt(pow(self.__x - other_point.GetX(), 2) + pow(self.__y - other_point.GetY(), 2))

  def GenerateSymmetricPoint(self):
    self.__x = -self.GetX()
    self.__y = -self.GetY()

  def TranslatePoint(self, translator_point):
    self.__x += translator_point.GetX() 
    self.__y += translator_point.GetY()      

File: JAVA/CartesianPoint/Point.java

import java.lang.Math;

public class Point {
	private double x_;
	private double y_;
	private double distance_;

	Point(double x, double y){
		x_ = x;
		y_ = y;
		distance_ = Math.sqrt(x * x + y * y);
	}

	public double GetX() {
		return x_;
	}
	public double GetY() {
		return y_;
	}
	public double GetDistanceFromOrigin() {
		return distance_;
	}
        
        public void SetX(double x_in) {
        	x_ = x_in;
        	SetDistanceFromOrigin();
        }
        public void SetY(double y_in) {
        	y_ = y_in;
        }
        public void SetDistanceFromOrigin() {
			distance_ = Math.sqrt(x_ * x_ + y_ * y_);
        }
        
        public void DisplayCoordinates() {
        	System.out.println("(" + String.valueOf(x_) + " " + String.valueOf(y_) + ")");
        }
        
        public double GetDistanceFromPoint(Point other_point) {
        	return Math.sqrt((x_ - other_point.x_) *(x_ - other_point.x_) + (y_ - other_point.y_) *(y_ - other_point.y_));
        }
        
        public void GenerateSymmetricPoint() {
        	x_ = -x_;
        	y_ = -y_;
        }
        public void TranslatePoint(Point other_point) {
        	x_ = x_ + other_point.x_;
        	y_ = y_ + other_point.y_;
        }
        
}
